lib/uart.toit

```diff
@@ -217,6 +219,64 @@ class Port implements reader.Reader:
       if flushed: return
       sleep --ms=1
 
+/**
+Extends the functionality of the UART Port on platforms that support configuratble RS232 devices. It allows setting
+and reading control lines.
+*/
+
+class ConfigurableDevicePort extends Port:
```

Comment:
Could we make this the `HostPort` class (someone has a better name?) or something similar?
The `Port` constructor that takes a device string could then be a factory constructor returning an instance of this class. (We would need a private constructor in the `Port` class to make this work).
This would make me more comfortable with having a new class.

Reply:
With this suggestion:
`port/HostPort := Port "/dev/ttyUSB1"` fails to compile.
`port/HostPort := (Port "/dev/ttyUSB1") as HostPort` compiles. 
Seems a bit convoluted

Reply:
We could still leave the `HostPort` constructor public. So if you actually wanted the `HostPort` directly, you can go there.

Just feels unnecessary to not give the host-functionality, just because you didn't go through the `HostPort` constructor.